x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/tool/BaseRunAsSuperuserCommand.java

```diff
@@ -0,0 +1,281 @@
+                ExitCodes.CONFIG,
+                "Multiple file realms are configured. "
+                    + "[file] is an internal realm type and therefore there can only be one such realm configured"
+            );
+        } else if (fileRealmSettings.size() == 1
+            && fileRealmSettings.entrySet().stream().anyMatch(s -> s.getValue().get("enabled").equals("false"))) {
+                throw new UserException(ExitCodes.CONFIG, "File realm must be enabled");
+            }
+        // Else it's either explicitly enabled, or not defined in the settings so it is implicitly enabled.
+    }
+
+    /**
+     * Checks that we can connect to the cluster and that the cluster health is not RED. It optionally handles
+     * retries as the file realm might not have reloaded the users file yet in order to authenticate our
+     * newly created file realm user.
+     */
+    private void checkClusterHealthWithRetries(Environment env, Terminal terminal, int retries) throws Exception {
+        CommandLineHttpClient client = clientFunction.apply(env);
+        final URL clusterHealthUrl = createURL(new URL(client.getDefaultURL()), "_cluster/health", "?pretty");
+        final HttpResponse response;
+        try {
+            response = client.execute("GET", clusterHealthUrl, username, password, () -> null, this::responseBuilder);
+        } catch (Exception e) {
+            throw new UserException(ExitCodes.UNAVAILABLE, "Failed to determine the health of the cluster. ", e);
+        }
+        final int responseStatus = response.getHttpStatus();
+        if (responseStatus != HttpURLConnection.HTTP_OK) {
+            if (responseStatus == HttpURLConnection.HTTP_UNAUTHORIZED && retries > 0) {
+                terminal.println(
+                    Terminal.Verbosity.VERBOSE,
+                    "Unexpected http status [401] while attempting to determine cluster health. Will retry at most "
+                        + retries
+                        + " more times."
+                );
+                Thread.sleep(1000);
+                retries -= 1;
+                checkClusterHealthWithRetries(env, terminal, retries);
+            } else {
+                throw new UserException(
+                    ExitCodes.DATA_ERROR,
+                    "Failed to determine the health of the cluster. Unexpected http status [" + responseStatus + "]"
+                );
+            }
+        } else {
+            final String clusterStatus = Objects.toString(response.getResponseBody().get("status"), "");
+            if (clusterStatus.isEmpty()) {
+                throw new UserException(
+                    ExitCodes.DATA_ERROR,
+                    "Failed to determine the health of the cluster. Cluster health API did not return a status value"
+                );
+            } else if ("red".equalsIgnoreCase(clusterStatus)) {
+                throw new UserException(ExitCodes.UNAVAILABLE, "Cluster health is currently RED.");
```

Comment:
Experience with setup-passwords suggests that new users don't know what to do with this sort of error.
If we can point to the logs directory, that's probably helpful (but depends on docker, etc).

Reply:
Agreed, I adjusted this to be more verbose and point to logs ( and docs as to where logs live )